File: Python/004.py
def ispalin(x):
	temp = x
	y = 0
	while(int(x)):
		y+=(x%10)
		y*=10
		x/=10
		x = int(x)
	y/=10
	if(temp == y):
		return 1
	else:
		return 0
ans = 0
for i in range(100,1000):
	for j in range(100,1000):
		calc = i*j
		if(ispalin(calc) and calc>ans):
			ans = calc
print(int(ans))

File: C++/004.cpp
#include<stdio.h>
int ispalin(int x){
	int temp;
	temp = x;
	int y=0;
	while(x){
		y+=(x%10);
		y*=10;
		x/=10;
	}
	y/=10;
	if(temp == y) return 1;
	else return 0;
}
int ans;
int main(){
	for(int i=100;i<1000;i++){
		for(int j=100;j<1000;j++){
			if(ispalin(i*j) && i*j>ans) ans=i*j;
		}
	}
	printf("%d",ans);
}